libs/ngx-nuts-and-bolts/src/lib/helpers/table-state/table-state.helper.ts

```diff
@@ -0,0 +1,204 @@
+import { ActivatedRoute, Router } from '@angular/router';
+import { isEqual } from 'lodash';
+import { Observable } from 'rxjs';
+import { distinctUntilChanged, map } from 'rxjs/operators';
+
+export type Sort = 'asc' | 'desc' | '';
+
+export interface IPageInfo {
+	pageIndex: number;
+	pageSize: number;
+	length?: number;
+}
+
+export interface ISortInfo {
+	sortingKey: string;
+	sortDirection: Sort;
+}
+
+export enum TableQueryParam {
+	PAGE_INDEX = 'pageIndex',
+	PAGE_SIZE = 'pageSize',
+	SORT_KEY = 'sortKey',
+	SORT_DIRECTION = 'sortDirection',
+	CUSTOM_FILTERS = 'filters',
+}
+
+/**
+ *
+ * Creates observable that emits when page information in route changes.
+ * @param {ActivatedRoute} route - Activated route instance used in component.
+ * @param {IPageInfo} [fallbackPaginationInfo] - Fallback values for pagination info.
+ */
+export function createPaginationObservable(
+	route: ActivatedRoute,
+	fallBackPaginationInfo: IPageInfo = { pageIndex: 0, pageSize: 10 }
+): Observable<IPageInfo> {
+	return route.queryParamMap.pipe(
+		map((paramMap) => {
+			const pageIndex = parseInt(
+				paramMap.get(TableQueryParam.PAGE_INDEX) ?? fallBackPaginationInfo.pageIndex.toString(),
+				10
+			);
+			const pageSize = parseInt(
+				paramMap.get(TableQueryParam.PAGE_SIZE) ?? fallBackPaginationInfo.pageSize.toString(),
+				10
+			);
+
+			return {
+				pageIndex,
+				pageSize,
+			};
+		}),
+		distinctUntilChanged(
+			(pageEvent1, pageEvent2) =>
+				pageEvent1.pageIndex === pageEvent2.pageIndex && pageEvent1.pageSize === pageEvent2.pageSize
+		)
+	);
+}
+
+/**
+ *
+ * Creates observable that emits when sorting information in route changes.
+ * @param {ActivatedRoute} route - Activated route instance used in component.
+ */
+export function createSortingObservable(route: ActivatedRoute): Observable<ISortInfo> {
```

Comment:
In some places we say `sort`, in some `sorting`. Pick one and be consistent, I don't have a preference, just pick one.